test/firtool/convention.fir

```diff
@@ -0,0 +1,23 @@
+; RUN: firtool %s --format=fir --parse-only --scalarize-top-module --scalarize-ext-modules --preserve-aggregate=all | FileCheck %s
```

Comment:
Consider a more fine-grained approach that has 4 `RUN` lines testing all behavior.